log.go

```diff
@@ -389,10 +389,11 @@ func (l *raftLog) term(i uint64) (uint64, error) {
 	}
 
 	// The valid term range is [index of dummy entry, last index].
-	dummyIndex := l.firstIndex() - 1
-	if i < dummyIndex || i > l.lastIndex() {
-		// TODO: return an error instead?
-		return 0, nil
+	if i+1 < l.firstIndex() {
```

Comment:
Out of curiosity, what would happen if we just removed this logic and let the code call `storage.Term`?

Reply:
I tried this, tests fail. This is probably due to the fact that `l.firstIndex() != storage.FirstIndex()`. The former takes into account the in-memory `unstable` portion of the log which patches the storage state.